ListRCConfiguration/handler.ts

```diff
@@ -0,0 +1,102 @@
+import * as express from "express";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorInternal,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+import { flow, pipe } from "fp-ts/lib/function";
+import { NonEmptyString, Ulid } from "@pagopa/ts-commons/lib/strings";
+import { retrievedRCConfigurationToPublic } from "@pagopa/io-functions-commons/dist/src/utils/rc_configuration";
+import { RCConfigurationModel } from "@pagopa/io-functions-commons/dist/src/models/rc_configuration";
+import { CosmosErrors } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { UserRCConfigurationModel } from "@pagopa/io-functions-commons/dist/src/models/user_rc_configuration";
+import { RequiredUserIdMiddleware } from "../middlewares/required_headers_middleware";
+import { RCConfigurationListResponse } from "../generated/definitions/RCConfigurationListResponse";
+
+interface IHandlerParameter {
+  readonly userId: NonEmptyString;
+}
+
+interface IListRCConfigurationHandlerParameter {
+  readonly rccModel: RCConfigurationModel;
+  readonly userRCConfigurationModel: UserRCConfigurationModel;
+}
+
+export const handleCosmosErrorResponse = (
+  error: CosmosErrors
+): IResponseErrorInternal =>
+  ResponseErrorInternal(
+    `Something went wrong trying to retrieve the configurations: ${JSON.stringify(
+      error
+    )}`
+  );
+
+export const listRCConfigurationHandler = ({
+  rccModel,
+  userRCConfigurationModel
+}: IListRCConfigurationHandlerParameter) => ({
+  userId
```

Comment:
I'd consider also to pass some parameters useful for pagination.
Today we have few configurations, but what if tomorrow we'll have a hundred configurations for every user? 
This is a pretty feasible situation, because a user is an aggregator for many different services.
We have 3 option here: 
* using continuation tokens for a "view more" or "infinite-scroll" pagination
* adopting the same tactic we adopted for the messages by sending `page_size`, `prev_id` and `next_id` to keep reference to the exact point we arrived during pagination
* using a classic pagination made by `page_size` and `page_number` to calculate offset and limit values
I think the most feasible approaches are the first and the last, because the second was specifically made for messages to allow the use of async iterators to skip some messages.

Reply:
The problem with pagination here is that the query that we do in the collection is an "IN" query using the configurationIds retrieved from the userRCConfiguration collection. That query returns all the documents associated to the list of configId and than we filter them out in the code to get the last version. How can we implement the pagination in this scenario?

Reply:
We will paginate the retrieval of that list. Instead of getting ALL the userRCConfigurations, we get them page by page, and then use the IN on the pages.

Reply:
Just to be clear, I don't think pagination is some desired sugar for this endpoint, I think pagination is really a MUST, and sooner or later we'll have to integrate it.